src/deepsparse/server/server.py

```diff
@@ -0,0 +1,159 @@
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from asyncio import get_running_loop
+from concurrent.futures import ThreadPoolExecutor
+from copy import deepcopy
+from typing import List
+
+import yaml
+
+import uvicorn
+from deepsparse.engine import Context, Scheduler
+from deepsparse.pipeline import Pipeline
+from deepsparse.server.config import EndpointConfig, ServerConfig
+from fastapi import FastAPI, UploadFile
+from starlette.responses import RedirectResponse
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def start_server(
+    config_path: str, host: str = "0.0.0.0", port: int = 5543, log_level: str = "info"
+):
+    """
+    Starts a FastAPI server with uvicorn with the configuration specified.
+
+    :param config_path: A yaml file with the server config. See :class:`ServerConfig`.
+    :param host: The IP address to bind the server to.
+    :param port: The port to listen on.
+    :param log_level: Log level given to python and uvicorn logging modules.
+    """
+    log_config = deepcopy(uvicorn.config.LOGGING_CONFIG)
+    log_config["loggers"][__name__] = {
+        "handlers": ["default"],
+        "level": log_level.upper(),
+    }
+
+    with open(config_path) as fp:
+        obj = yaml.safe_load(fp)
+    server_config = ServerConfig(**obj)
+    _LOGGER.info(f"Using config: {repr(server_config)}")
+
+    app = _build_app(server_config)
+
+    uvicorn.run(
+        app,
+        host=host,
+        port=port,
+        log_level=log_level,
+        log_config=log_config,
+        # NOTE: only want to have 1 server process so models aren't being copied
+        workers=1,
+    )
+
+
+def _build_app(server_config: ServerConfig) -> FastAPI:
+    context = Context(
+        num_cores=server_config.num_cores,
+        num_streams=server_config.num_workers,
+        scheduler=Scheduler.multi_stream,
+    )
+
+    _LOGGER.info(f"Built context: {repr(context)}")
+
+    app = FastAPI()
+
+    @app.get("/", include_in_schema=False)
+    def _home():
+        return RedirectResponse("/docs")
+
+    @app.get("/config", tags=["general"], response_model=ServerConfig)
+    def _info():
+        return server_config
+
+    @app.get("/ping", tags=["general"], response_model=bool)
+    @app.get("/health", tags=["general"], response_model=bool)
+    @app.get("/healthcheck", tags=["general"], response_model=bool)
+    @app.get("/status", tags=["general"], response_model=bool)
+    def _health():
+        return True
+
+    app._deepsparse_pool = ThreadPoolExecutor(max_workers=context.num_streams)
+
+    add_invocations_endpoint = len(server_config.endpoints) == 1
```

Comment:
the logic for re-routing the endpoint path with /invocations for sagemaker support was taken out. need to get some functionality for this added back in and update the existing sagemaker documentation/dockerfile if the location of where to set integration has changed. ping if any questions

Reply:
Can you expand? The server will add an /invocations endpoint if there's only 1 endpoint, was there something else I missed bringing over?

Reply:
missing functionality expanded on in thread below. and would be good to add/keep in a pathway for specifying deployment scenario (can be above the server config level, like CLI) because for something like multi-model sagemaker we'll want to know and detect when to add in the extra sagemaker routes. having knowledge of integrations is also important for choosing which logger integration(s) to use/add in

additionally, don't think it's necessary to always add in `invocations` for single model servers since for non sagemaker deployments it will just be extra spam in the auto docs/swagger ui

Reply:
Okay I added integration field back in, and now the logic for sagemaker should match what was there before